Controller/Notify_Handler.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Controller;
+
+use \Wirecard\Oxid\Extend\Order;
+use \Wirecard\Oxid\Core\Payment_Method_Factory;
+use \Wirecard\Oxid\Model\Transaction;
+
+use \Wirecard\PaymentSdk\Response\Response;
+use \Wirecard\PaymentSdk\TransactionService;
+use \Wirecard\PaymentSdk\Exception\MalformedResponseException;
+use \Wirecard\PaymentSdk\Response\SuccessResponse;
+
+use \OxidEsales\Eshop\Application\Model\Order as Oxid_Order;
+use \OxidEsales\Eshop\Application\Controller\FrontendController;
+use \OxidEsales\Eshop\Core\Registry;
+use \OxidEsales\Eshop\Core\Field;
+
+use \Psr\Log\LoggerInterface;
+use \InvalidArgumentException;
+
+/**
+ * Notify handler class.
+ *
+ * Handle Payment SDK notifications.
+ */
+class Notify_Handler extends FrontendController
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $_oLogger;
+
+    /**
+     * NotifyHandler constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->_oLogger = Registry::getLogger();
+    }
+
+    /**
+     * Request handling function.
+     *
+     * @return void
+     */
+    public function handleRequest()
+    {
+        $sPaymentName = Registry::getRequest()->getRequestParameter('pmt');
+        $oPaymentMethod = Payment_Method_Factory::create($sPaymentName);
+        $oConfig = $oPaymentMethod->getConfig();
+        $sPostData = file_get_contents('php://input');
+
+        try {
+            $oService = new TransactionService($oConfig, $this->_oLogger);
+            $oNotificationResponse = $oService->handleNotification($sPostData);
+        } catch (InvalidArgumentException $exception) {
+            $this->_oLogger->error($exception->getMessage());
+            return;
+        } catch (MalformedResponseException $exception) {
+            $this->_oLogger->error($exception->getMessage());
+            return;
+        }
+
+        // Return the response or log errors if any happen.
+        if ($oNotificationResponse instanceof SuccessResponse && $oNotificationResponse->isValidSignature()) {
+            $this->_onNotificationSuccess($oNotificationResponse);
+        } else {
+            $this->_onNotificationError($oNotificationResponse);
+        }
+    }
+
+    /**
+     * Handles success notifications
+     *
+     * @param Response $oResponse
+     *
+     * @return void
+     */
+    private function _onNotificationSuccess(Response $oResponse)
+    {
+        $this->_oLogger->debug('Notification response: ' . $oResponse->getRawData());
+        $aData = $oResponse->getData();
+
+        $sParentTransactionId = $aData['parent-transaction-id'];
```

Comment:
Why are you using the parent transaction id?

Reply:
there are getters in the response for that values

Reply:
used getters if possible

Reply:
resolved as discussed